submissions/tests/test_team_api.py

```diff
@@ -489,3 +502,62 @@ def test_reset_scores_error(self, mock_individual_reset):
         team_submission = self._make_team_submission(create_submissions=True)
         with self.assertRaises(TeamSubmissionInternalError):
             team_api.reset_scores(team_submission.uuid)
+
+    def test_get_team_submission_for_student(self):
+        # Create some Team Submissions without Individual Submissions
+        team_1_submission = self._make_team_submission(team_id=TEAM_1_ID)
+        team_2_submission = self._make_team_submission(team_id=TEAM_2_ID)
+
+        # Make individual submissions
+        user_to_team_submission = {
+            self.user_1: team_1_submission,
+            self.user_2: team_1_submission,
+            self.user_3: team_2_submission,
+            self.user_4: team_2_submission,
+        }
+        user_to_individual_submission = {
+            user: self._make_individual_submission(
+                self.anonymous_user_id_map[user],
+                team_submission=user_to_team_submission[user]
+            ) for user in user_to_team_submission
+        }
+
+        # Assert that each student item returns the correct team submission
+        for user in user_to_individual_submission:
+            individual_submission = user_to_individual_submission[user]
+            student_item_dict = individual_submission.student_item.student_item_dict
+            actual_team_submission = team_api.get_team_submission_for_student(student_item_dict)
+
+            self.assertEqual(
+                str(user_to_team_submission[user].uuid),
+                actual_team_submission['team_submission_uuid']
+            )
+
+    def test_get_team_submission_for_student__no_team_submission(self):
```

Comment:
Nit: Is the double underscore in the function name intentional?

Reply:
yeah, it's just to separate it from the repeated prefix